nvdaHelper/remote/sysListView32.cpp

```diff
@@ -59,3 +59,46 @@ error_status_t nvdaInProcUtils_sysListView32_getColumnContent(handle_t bindingHa
 	*text = SysAllocString(lvItem.pszText);
 	return 0;
 }
+
+error_status_t nvdaInProcUtils_sysListView32_getColumnLocation(handle_t bindingHandle, const unsigned long windowHandle, int item, int subItem, RECT* location) {
+	RECT localRect {
+		// Returns the bounding rectangle of the entire item, including the icon and label.
+		LVIR_LABEL,  // left
+		// According to Microsoft, top should be the one-based index of the subitem.
+		// However, indexes coming from LVM_GETCOLUMNORDERARRAY are zero based.
+		subItem // top
+	};
+	if (!SendMessage((HWND)UlongToHandle(windowHandle), LVM_GETSUBITEMRECT, (WPARAM)item, (LPARAM)&localRect)) {
+		LOG_DEBUGWARNING(L"LVM_GETSUBITEMRECT failed");
+		return 1;
+	}
+	*location = localRect;
+	return 0;
+}
+
+error_status_t nvdaInProcUtils_sysListView32_getColumnHeader(handle_t bindingHandle, const unsigned long windowHandle, int subItem, BSTR* text) {
+	LVCOLUMN lvColumn = {0};
+	lvColumn.mask = LVCF_TEXT;
+	lvColumn.iSubItem = subItem;
+	lvColumn.cchTextMax = CBEMAXSTRLEN;
+	wchar_t textBuf[CBEMAXSTRLEN]{}; // Ensure that the array initialised with all zero values ('\0')
+	lvColumn.pszText= textBuf;
+	if (!SendMessage((HWND)UlongToHandle(windowHandle), LVM_GETCOLUMN, (WPARAM)subItem, (LPARAM)&lvColumn)) {
+		LOG_DEBUGWARNING(L"LVM_GETCOLUMN failed");
+		return 1;
+	}
+	if(!lvColumn.pszText) {
+		LOG_DEBUGWARNING(L"LVM_GETCOLUMN didn't retrieve any text");
+		return 1;
+	}
+	*text = SysAllocString(lvColumn.pszText);

```

Comment:
I'd prefer if `SysAllocStringLen` was used to ensure the buffer size isn't exceeded. Is the size/number of chars returned in the `lvColumn` struct?

Reply:
This is not possible because cchTextMax won't be changed to the actual number of characters in the buffer. It would result in many null characters in the resulting BSTR.

Reply:
Ok, could you add a comment on that please.